### DISCUSSION TITLE
how to create new table with CRUD for blazor?

### DISCUSSION THREAD

**Discussion by TejpallPathak** [ServiceStack/Discuss#15] [Q&A]
i am learning and saw sample but can you please help me how to create new table in db for blazor like booking? and also how to make CRUD pages and methods?

**Comment by mythz** [comment]
Here's the source code for the [blazor-wasm](https://github.com/NetCoreTemplates/blazor-wasm) template.

It uses OrmLite (https://docs.servicestack.net/ormlite/) to create the tables which it does in: [MyApp/Configure.Db.cs](https://github.com/NetCoreTemplates/blazor-wasm/blob/main/MyApp/Configure.Db.cs#L21)

```csharp
if (db.CreateTableIfNotExists<Booking>())
{
    db.CreateBooking("First Booking!", RoomType.Queen, 10, 100, "employee@email.com");
    db.CreateBooking("Booking 2", RoomType.Double, 12, 120, "manager@email.com");
    db.CreateBooking("Booking the 3rd", RoomType.Suite, 13, 130, "employee@email.com");
}
```

The Create/Edit Bookings pages are located at: [MyApp.Client/Pages/BookingsCrud](https://github.com/NetCoreTemplates/blazor-wasm/tree/main/MyApp.Client/Pages/BookingsCrud)

They create/edit Bookings using only the AutoQuery CRUD APIs defined in: [MyApp.ServiceModel/Bookings.cs](https://github.com/NetCoreTemplates/blazor-wasm/blob/main/MyApp.ServiceModel/Bookings.cs)

[AutoQuery](https://docs.servicestack.net/autoquery-rdbms) and [AutoQuery CRUD](https://docs.servicestack.net/autoquery-crud) is ServiceStack's solution for creating CRUD APIs with just the Request DTO definition, all defined in [Bookings.cs](https://github.com/NetCoreTemplates/blazor-wasm/blob/main/MyApp.ServiceModel/Bookings.cs)

TalentBlazor is a new App (deployed to https://talent.locode.dev) which exclusively uses AutoQuery for its CRUD APIs whose source code is located at:

https://github.com/NetCoreApps/TalentBlazor
